### ISSUE TITLE
Should map_saver have raw mode ?

### ISSUE THREAD

**Issue by jingedawang** [ros-planning/navigation#830]
There are many circumstances that raw map is needed, such as localization and path planning. But current version of map_saver can only save trinary map.

I am wondering if ROS organization have a plan to add this feature ? Or if I add this feature, are you willing to merge it ? I can guarantee backward compatibility.

**Comment by jingedawang** [comment]
@mikeferguson

**Comment by DLu** [comment]
I don't believe anyone is currently working on this. We are willing to merge pull requests.

**Comment by jingedawang** [comment]
I have implemented it. Please have a look. #832

**Comment by skpro19** [comment]
@jingedawang what do you mean by `trinary` map?

**Comment by jingedawang** [comment]
> @jingedawang what do you mean by `trinary` map?

There are three values in the final map, which stands for **free**, **occupied**, **unknown** respectively.

**Comment by skpro19** [comment]
Makes sense.
